gcloud-java-storage/src/main/java/com/google/gcloud/storage/testing/FakeStorageRpc.java

```diff
@@ -117,18 +121,20 @@ public StorageObject create(StorageObject object, InputStream content, Map<Optio
       if (!so.getName().startsWith(prefix)) {
         continue;
       }
-      int nextSlash = so.getName().indexOf("/", prefix.length());
-      if (nextSlash >= 0) {
-        String folderName = so.getName().substring(0, nextSlash + 1);
-        if (folders.containsKey(folderName)) {
+      if (delimiter != null) {
+        int nextSlash = so.getName().indexOf(delimiter, prefix.length());
+        if (nextSlash >= 0) {
+          String folderName = so.getName().substring(0, nextSlash + 1);
+          if (folders.containsKey(folderName)) {
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Deeply nested if..then statements are hard to read](https://www.codacy.com/app/mziccard/gcloud-java/file/2414371026/issues/source?bid=3123107&fileBranchId=3236493#l128)